models/device.py

```diff
@@ -163,9 +169,22 @@ def generate_uuid(self):
         # pass the certificate validation in OpenVPN This either means that
         # the RFC counts a final NULL byte as part of the CN or that the
         # OpenVPN/OpenSSL implementation has a bug.
-        rand_string = os.urandom(31)
-        return binascii.hexlify(rand_string)
+        return binascii.hexlify(os.urandom(31))
+
+    def register(self, app_name, uuid):
+        user_id = self.token.get_user_id()
+        application = self.application.get(app_name)
+        api_key = self.base_request.request('/application/{0}/generate-api-key'.format(application['id']), 'POST', endpoint=self.settings.get('pine_endpoint'))
+
+        data = {
+            'user': user_id,
+            'application': application['id'],
+            'device_type': application['device_type'],
+            'registered_at': (datetime.utcnow() - datetime.utcfromtimestamp(0)).total_seconds(),
+            'uuid': uuid
+        }
 
-    # NOT YET IMPLEMENTED, WAITING FOR AUTH
-    #def register(self, app_name, uuid):
+        if api_key:
+            data['apikey'] = api_key
 
+        return self.base_request.request('device', 'POST', data=data, endpoint=self.settings.get('pine_endpoint'))
```

Comment:
How is `pine_endpoint` different from `api_endpoint`? is it `api_endpoint + '/ewa'`? then maybe the config should better have `pine_prefix` instead? Or is it mirroring the model of Node SDK?